mmv1/products/vertexai/FeatureOnlineStore.yaml

```diff
@@ -41,65 +43,183 @@ description: Vertex AI Feature Online Store provides a centralized repository fo
 autogen_async: false
 examples:
   - !ruby/object:Provider::Terraform::Examples
-    name: vertex_ai_feature_online_store
-    primary_resource_id: feature_online_store
+    name: 'vertex_ai_featureonlinestore_with_bigtable'
+    primary_resource_id: 'featureonlinestore'
+    primary_resource_name: "fmt.Sprintf(\"terraform%s\",
+      context[\"random_suffix\"\
+      ])"
     vars:
-      name: example_feature_online_store
+      name: 'terraform2'
+      project: 'appeng-flex'
+    test_env_vars:
+      org_id: :ORG_ID
+      billing_account: :BILLING_ACCT
+    ignore_read_extra:
+      - 'force_destroy'
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'vertex_ai_featureonlinestore_with_beta_fields_optimized'
+    primary_resource_id: 'featureonlinestore'
+    primary_resource_name: "fmt.Sprintf(\"terraform%s\",
+      context[\"random_suffix\"\
+      ])"
+    vars:
+      name: 'terraform'
+      project: 'appeng-flex'
+    test_env_vars:
+      org_id: :ORG_ID
+      billing_account: :BILLING_ACCT
+    ignore_read_extra:
+      - 'force_destroy'
+    min_version: beta
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'vertex_ai_featureonlinestore_with_beta_fields_bigtable'
+    primary_resource_id: 'featureonlinestore'
+    primary_resource_name: "fmt.Sprintf(\"terraform%s\",
+      context[\"random_suffix\"\
+      ])"
+    vars:
+      name: 'terraform3'
+      project: 'appeng-flex'
+    test_env_vars:
+      org_id: :ORG_ID
+      billing_account: :BILLING_ACCT
+    ignore_read_extra:
+      - 'force_destroy'
+    min_version: beta
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  pre_delete: templates/terraform/pre_delete/vertex_ai_force_delete.go.erb
+virtual_fields:
+  - !ruby/object:Api::Type::Boolean
+    name: 'force_destroy'
+    description:
+      'If set to true, any FeatureViews and Features for this FeatureOnlineStore will also be deleted.'
+    default_value: false
 parameters:
   - !ruby/object:Api::Type::String
     name: region
     description: The region of feature online store. eg us-central1
     url_param_only: true
     immutable: true
-    required: true
+    default_from_api: true
 properties:
   - !ruby/object:Api::Type::String
-    name: name
-    description: The resource name of the Feature Online Store.
+    name: 'name'
+    description:
+      The name of the featureOnlinestore. This value may be up to 60 characters, and
+      valid characters are [a-z0-9_]. The first character cannot be a number.
     immutable: true
-    required: true
+    url_param_only: true
```

Comment:
Do you know if the configuration for `name` needs to change as you've suggested here? My understanding is that the previous config was working because the `name` can be provided in the request body, and it is returned in the response body, so it is arguably a simpler approach.

Reply:
I've done this so that it is consistent with the legacy FeatureStore which has similar url structure and json request object. If the existing one is simpler and easy to understand, I can change it too. Pls let me know.

Reply:
The precedence for vertexai is a bit fragmented unfortunately, with Endpoint being the primary exception to the pattern you mention. However, since most of the other resources do use `url_param_only` and `pattern`, I think it makes sense to keep them here.

However, are you sure we should remove `required`? What happens if a name is not specified?

Reply:
Correct! The REST API documentation doesn't explicitly mention it as required field but when I try to run without this field specified, it throws an error. So, updated to include `required : true` now